dashboard/modules/reporter/profile_manager.py

```diff
@@ -131,6 +144,125 @@ async def cpu_profile(
         )
         stdout, stderr = await process.communicate()
         if process.returncode != 0:
-            return False, _format_failed_pyspy_command(cmd, stdout, stderr)
+            return False, _format_failed_profiler_command(
+                cmd, self.profiler, stdout, stderr
+            )
         else:
             return True, open(profile_file_path, "rb").read()
+
+
+class MemoryProfilingManager:
+    def __init__(self, profile_dir_path: str):
+        self.profile_dir_path = Path(profile_dir_path) / "memray"
```

Comment:
You should validate profile_dir_path exists first? I think if profile_dir_path doesn't exist, the next mkdir will fail?